faunadb-java/src/main/java/com/faunadb/client/FaunaClient.java

```diff
@@ -455,6 +456,9 @@ private void handleQueryErrors(int statusCode, String body) {
           if (ex instanceof ConnectException || ex instanceof TimeoutException) {
               throw new UnavailableException(ex.getMessage(), ex);
           }
+          if (ex instanceof CompletionException && ex.getMessage().contains("too many concurrent streams")) {
+            throw new BadRequestException("the maximum number of streams has been reached for this client");
```

Comment:
Do we want this error to be precisely at the 100 mark? If so, we need to synchronize access to the portion of the JDK's http client that initiates an http2 stream. Otherwise, a thread race could cause it to open more than 100 streams. If we don't care for precision, it's okay then.

Reply:
This is a good point.
AFAIK we do not care about the precision, as the error would create more than 100 streams and not less.

I am fine adding synchronization if someone really wants it however :)